charts/ColorScaleConfig.ts

```diff
@@ -2,28 +2,70 @@ import { observable } from "mobx"
 
 import { Color } from "./Color"
 
+export enum ColorScaleBinningStrategy {
+    equalInterval = "equalInterval",
+    quantiles = "quantiles",
+    ckmeans = "ckmeans",
+    manual = "manual"
+}
+
+export const colorScaleBinningStrategyLabels: Record<
+    ColorScaleBinningStrategy,
+    string
+> = {
+    equalInterval: "Equal-interval",
+    quantiles: "Quantiles",
+    ckmeans: "Ckmeans",
+    manual: "Manual"
+}
+
 export class ColorScaleConfigProps {
-    // Key for a colorbrewer scheme, may then be further customized
-    @observable.ref baseColorScheme?: string
+    // Color scheme
+    // ============
+
+    /** Key for a colorbrewer scheme */
+    @observable baseColorScheme?: string
+
+    /** Reverse the order of colors in the color scheme (defined by `baseColorScheme`) */
+    @observable colorSchemeInvert?: true = undefined
 
-    // Minimum value shown on map legend
-    @observable.ref colorSchemeMinValue?: number
-    @observable colorSchemeValues: number[] = []
-    @observable colorSchemeLabels: (string | undefined)[] = []
-    @observable.ref isManualBuckets?: true = undefined
-    @observable.ref equalSizeBins?: true = undefined
+    // Numeric bins
+    // ============
 
-    // Whether to reverse the color scheme on output
-    @observable.ref colorSchemeInvert?: true = undefined
-    @observable.ref customColorsActive?: true = undefined
+    /** The strategy for generating the bin boundaries */
+    @observable binningStrategy: ColorScaleBinningStrategy =
+        ColorScaleBinningStrategy.ckmeans
+    /** The *suggested* number of bins for the automatic binning algorithm */
+    @observable binningStrategyBinCount?: number
 
-    // e.g. ["#000", "#c00", "#0c0", "#00c", "#c0c"]
+    // Iff the binningStrategy is `manual`, then overrides are specified below.
+    // Otherwise, the overrides are ignored.
```

Comment:
Should overrides be grouped together into a `manualOverrides` object?

Reply:
I've edited the comments now, only 2 of the 3 properties are actually affected – values and labels. Minimum value is still used even if the strategy is non-manual.

But noticed an existing bug looking into this – the custom numeric labels were applied even when automatic binning is used.

Re nesting objects: we usually avoid it in the rest of the config, and I don't feel it's necessary here. Basically anything beginning with `custom` is a manual override, which are most properties in this config object.